### PR TITLE
Bump 2.21.2 - Apply imds patch & upgrade to enumerated cherry-pick patches

### PR DESCRIPTION
**Pull request by matthewfala**
Signed-off-by: Matthew Fala <falamatt@amazon.com>

*Issue #, if available:*
This patch resolves the IMDSv1 fallback error mentioned in the following issue: #259.

*Description of changes:*
Introduces a patch that allows for graceful IMDSv1 fallback if IMDSv2 is not available due to the hop limit error.
Introduces network timeout dependency feature to Fluent Bit which we are working on getting merged upstream. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -41,9 +41,20 @@ ENV PATH ${PATH}:/home/.gimme/versions/go1.17.linux.arm64/bin:/home/.gimme/versi
...
 
 WORKDIR /tmp/fluent-bit-$FLB_VERSION/
-RUN git clone https://github.com/matthewfala/fluent-bit.git /tmp/fluent-bit-$FLB_VERSION/
+RUN git clone https://github.com/fluent/fluent-bit.git /tmp/fluent-bit-$FLB_VERSION/
 WORKDIR /tmp/fluent-bit-$FLB_VERSION/build/
-RUN git fetch origin && git checkout 1_8_9_patch && git status
+RUN git fetch --all --tags && git checkout tags/v${FLB_VERSION} -b v${FLB_VERSION} && git describe --tags
+
+RUN git config --global user.email "falamatt@amazon.com" \
+  && git config --global user.name "Matthew Fala"
```
Comment (matthewfala) [inline Dockerfile:49]: I was getting errors when an email and name were not configured on git cherry picks. I entered my email for now, however if there is a better email to use like one that we use for the observability team or firelens team, I can change it to that.

File: Dockerfile [hunk 2]
```diff
@@ -41,9 +41,20 @@ ENV PATH ${PATH}:/home/.gimme/versions/go1.17.linux.arm64/bin:/home/.gimme/versi
...
+
+# Apply Fluent Bit patches to base version
+COPY AWS_FLB_CHERRY_PICKS \
+  /AWS_FLB_CHERRY_PICKS
+
+RUN cat /AWS_FLB_CHERRY_PICKS | \
+  xargs -l bash -c 'git fetch $0 $1 && git cherry-pick $2'
```
Comment (matthewfala) [inline Dockerfile:56]: This cherry picks commits enumerated in added AWS_FLB_CHERRY_PICKS file.

File: AWS_FLB_CHERRY_PICKS [hunk 3]
```diff
@@ -0,0 +1,5 @@
+https://github.com/krispraws/fluent-bit.git tls_net_read_fix 8d1cfeb5ba830b360fe6e1228190ed900842a3ea
+https://github.com/zhonghui12/fluent-bit.git custom-1.8.7 30fc6305695623cbc95d51df07ae185dfec8bff2
+https://github.com/matthewfala/fluent-bit.git immutable-1.8-imds-r2-cherry-picks-r2 e44f8789d605f9ab19a212b4139bc7fec8d0b530
+https://github.com/matthewfala/fluent-bit.git immutable-1.8-imds-r2-cherry-picks-r2 30b00b295db9e3ee646a0262a07ca53cf0025b58
+https://github.com/matthewfala/fluent-bit.git immutable-1.8-imds-r2-cherry-picks-r2 890915050ef2353a06c7d3e079e4d2ce821fc30b
```
Comment (matthewfala) [inline AWS_FLB_CHERRY_PICKS:5]: Commit 1 e44f8789: The sync network timeout
Commit 2 30b00b29:  Default to old sync network code when timeout is not used
Commit 3 89091505:  IMDS v1 fallback patch

File: Dockerfile (hunk 2 above)
Comment (PettitWesley) [reply Dockerfile:56]: Can we just use the old strategy of having a specific branch? It makes it easy to check that we did everything correctly and released exactly the right code- it's simpler to code review.

File: Dockerfile (hunk 1 above)
Comment (hossain-rayhan) [reply Dockerfile:49]: Can we use our Firelens team email?

File: Dockerfile (hunk 2 above)
Comment (hossain-rayhan) [reply Dockerfile:56]: Yea, I also prefer to cut a release from a single branch.

Comment (PettitWesley) [reply Dockerfile:56]: Discussed with Matt, I am okay with using this approach after the addition of the `git log` statement below which will print the commits that were added to the final branch that is built

File: Dockerfile (hunk 1 above)
Comment (matthewfala) [reply Dockerfile:49]: What is our firelens team email?

Comment (zhonghui12) [reply Dockerfile:49]: aws-firelens@amazon.com

File: Dockerfile (hunk 2 above)
Comment (zhonghui12) [reply Dockerfile:56]: In this way, we should remember to check the `git log` result in our pipeline console even if _build_ stage succeeds.

File: Dockerfile (hunk 1 above)
Comment (matthewfala) [reply Dockerfile:49]: great! Updating the pr.

File: Dockerfile (hunk 2 above)
Comment (matthewfala) [reply Dockerfile:56]: Here's what to look out for:
 ```
 => => # Cherry Pick Patch Summary:                                                                                                                                                                    
 => => # Commit 1 -- 88571668 tls: Issue #4098 Fix error handling for OpenSSL apis                                                                                                                     
 => => # Commit 2 -- dacbff5d downgrade mbedtls to 2.24.0                                                                                                                                              
 => => # Commit 3 -- cdccb530 io: revised support timeout on synchronous calls                                                                                                                         
 => => # Commit 4 -- 028536b8 io: default to old net_io_read                                                                                                                                           
 => => # Commit 5 -- 60098b3d aws: revised imds fallback to v1 if token request fails
```

We need to make sure that these commits match the commits listed here:
https://github.com/aws/aws-for-fluent-bit/blob/8a1989e7d20618609f213b9528d154b842b44352/AWS_FLB_CHERRY_PICKS

File: AWS_FLB_CHERRY_PICKS (hunk 3 above)
Comment (matthewfala) [reply AWS_FLB_CHERRY_PICKS]: Added documentation directly to the cherry-pick enumeration file.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,20 @@
...
+This release includes:
+* An Amazon Linux 2 Base
+* Fluent Bit [1.8.9](https://fluentbit.io/announcements/v1.8.9/)
+* Amazon CloudWatch Logs for Fluent Bit 1.6.3
+* Amazon Kinesis Streams for Fluent Bit 1.8.0
+* Amazon Kinesis Firehose for Fluent Bit 1.6.1
+
+Compared to `2.21.1` this release includes the following fixes for AWS customers that we are working on getting accepted upstream:
+* Bug - Resolve IMDSv1 fallback error introduced in 2.21.0 [aws-for-fluent-bit:259](https://github.com/aws/aws-for-fluent-bit/issues/259)
+
```
Comment (PettitWesley) [inline CHANGELOG.md:13]: Need to also add the CW go plugin change right?

**Review by zhonghui12 — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (matthewfala) [reply CHANGELOG.md:14]: Done

Comment (PettitWesley) [reply CHANGELOG.md:14]: Now the "following fixes for AWS customers that we are working on getting accepted upstream" language is technically inaccurate. its only true for one of them

**Review by PettitWesley — APPROVED** [review]

**Pull request merged**